doc/kernel/services/interrupts.rst

```diff
@@ -173,10 +175,14 @@ The kernel addresses such use-cases by allowing interrupts with critical
 latency constraints to execute at a priority level that cannot be blocked
 by interrupt locking. These interrupts are defined as
 *zero-latency interrupts*. The support for zero-latency interrupts requires
-:kconfig:option:`CONFIG_ZERO_LATENCY_IRQS` to be enabled. In addition to that, the
-flag :c:macro:`IRQ_ZERO_LATENCY` must be passed to :c:macro:`IRQ_CONNECT` or
-:c:macro:`IRQ_DIRECT_CONNECT` macros to configure the particular interrupt
-with zero latency.
+:kconfig:option:`CONFIG_ZERO_LATENCY_IRQS` to be enabled. Any interrupts
+configured as zero-latency must also be declared as :ref:`direct ISRs
+<direct_isrs>` (and must not use the :c:macro:`ISR_DIRECT_PM` in them), since
+regular ISRs interact with the kernel. In addition to that, the flag
```

Comment:
The flag `IRQ_ZERO_LATENCY` is ARM specific, its not required nor possible to pass this flag for other archs. https://github.com/zephyrproject-rtos/zephyr/blob/1207880355fa2879877a2f06fbfe6ccd365603bb/include/zephyr/arch/arm/irq.h#L89

Reply:
There's a note that states that this only works for Cortex-M at the moment, but the flag will (and must) be moved out to common code if we add other architectures? what would you suggest we write?

Reply:
ah, I see, I thought this would be supported by more archs, atleast conceptually it should, disregard my comment :)